### PR TITLE
Teachercon partner registration bug bash feedback

### PR DESCRIPTION
**Pull request by mehalshah**

### CODE REVIEW

**Comment by mehalshah** [comment]
https://trello.com/c/rXxuCWCy/515-bug-bash-feedback-regional-partners-registering-for-teachercon

**Comment by mehalshah** [comment]
- [ ] Error messages for signed out users, non regional partner program managers, and non G3 partners
- [ ] Copy change in confirmation mail
- [ ] Allow multiple submissions per regional partner

File: dashboard/app/controllers/pd/teachercon1819_registration_controller.rb [hunk 1]
```diff
@@ -61,8 +61,13 @@ def new
   end
 
   def partner
-    unless current_user.regional_partners.count > 0
-      render :unauthorized
+    if current_user.nil? || !current_user.permission?(UserPermission::PROGRAM_MANAGER)
```
Comment (aoby) [inline dashboard/app/controllers/pd/teachercon1819_registration_controller.rb]: how about 
```ruby
unless current_user && current_user.permission?(UserPermission::PROGRAM_MANAGER)
```
?

Comment (mehalshah) [reply dashboard/app/controllers/pd/teachercon1819_registration_controller.rb]: Sure - or better yet
`unless current_user.try(:permission?, UserPermission::PROGRAM_MANAGER)`

File: dashboard/test/controllers/pd/teachercon1819_registration_controller_test.rb [hunk 2]
```diff
@@ -92,4 +96,61 @@ class Pd::Teachercon1819RegistrationControllerTest < ::ActionController::TestCas
...
+  # Partner not signed in should be redirected to please_sign_in
+  test 'Not signed in should be directed to please_sign_in page' do
+    sign_out(@teacher)
+
+    get :partner
+    assert_template('please_sign_in')
+  end
+
+  # Partner should redirect to please_sign_in for a non program manager
+  test 'Partner non-partner should redirect to please_sign_in page' do
```
Comment (clareconstantine) [inline dashboard/test/controllers/pd/teachercon1819_registration_controller_test.rb]: should this be partner non-program manager?

**Pull request merged**